python/lsst/ts/mtreflector/controller.py

```diff
@@ -0,0 +1,213 @@
+import asyncio
+import functools
+import logging
+import types
+
+# Hide mypy error `Module "labjack" has no attribute "ljm"`.
+from labjack import ljm  # type: ignore
+from lsst.ts.xml.enums.MTReflector import MTReflectorStatus
+
+# Time limit for communicating with the LabJack (seconds).
+COMMUNICATION_TIMEOUT = 5
+# Sleep time before trying to reconnect (seconds).
+RECONNECT_WAIT = 60
+
+
+class Controller:
+    """Implement MTReflector controller.
+
+    Parameters
+    ----------
+    config : `types.SimpleNamespace`
+        The configuration object.
+    log : `logging.Logger`
+        The log object.
+    simulation_mode : `int`
+        Controller in simulation mode?
+    Attributes
+    ----------
+    handle : `None` or `int`
+        The labjack ljm handle.
+    state : `lsst.ts.xml.MTReflector.MTReflectorStatus`
+        The current state of the reflector.
+    device_type : `int`
+        The labjack device type.
+    connection_type : `int`
+        The connection type to the labjack.
+    identifier : `str`
+        The host name of the labjack.
+    simulation_mode : `int`
+        Controller in simulation mode.
+    log : `logging.Logger`
+        The log object.
+    config : `types.SimpleNamespace`
+        The config object.
+    fake_value : `None` or `int`
+        The fake value used by simulation mode.
+    """
+
+    def __init__(
+        self,
+        config: types.SimpleNamespace,
+        log: logging.Logger | None,
+        simulation_mode: int,
+    ) -> None:
+        self.handle: None | int = None
+        self.state: MTReflectorStatus = MTReflectorStatus.UNKNOWN
+        self.device_type: int = ljm.constants.dtT4
+        self.connection_type: int = ljm.constants.ctTCP
+        self.identifier: None | str = None
+        self.simulation_mode: int = simulation_mode
+        self.log: logging.Logger = (
+            log if log is not None else logging.getLogger(__name__)
+        )
+        self.config: types.SimpleNamespace = config
+        self.fake_value: None | float = None
+
+    @property
+    def connected(self) -> bool:
+        """Connected to labjack?"""
+        return self.handle is not None
+
+    def configure(self) -> None:
+        """Configure the controller."""
+        if not self.simulation_mode:
+            self.identifier = self.config.identifier
+        else:
+            self.identifier = ljm.constants.DEMO_MODE
+
+    async def connect(self) -> None:
+        """Connect to the labjack."""
+        self.configure()
+        loop: asyncio.AbstractEventLoop = asyncio.get_event_loop()
+        try:
+            self.handle = await loop.run_in_executor(
+                None,
+                functools.partial(
+                    ljm.open,
+                    deviceType=self.device_type,
+                    connectionType=self.connection_type,
+                    identifier=self.identifier,
+                ),
+            )
+            await loop.run_in_executor(
```

Comment:
There are several repetitions of this ruin_in_executor constructs to write to the LabJack. Please extract to a method with parameters and call that everywhere. I would include getting the `loop` and the error handling in that method as well.